File: sources/java/_04_array_sum.java
// task 04 array_sum — expected output: 499999500000
// build: javac _04_array_sum.java    run: java _04_array_sum
// build (graalvm native-image): native-image -O2 _04_array_sum    run: ./_04_array_sum

public class _04_array_sum {
    public static void main(String[] args) {
        final int n = 1000000;
        long[] array = new long[n];
        for (int i = 0; i < n; i++) {
            array[i] = i;
        }
        long total = 0;
        for (int i = 0; i < n; i++) {
            total += array[i];
        }
        System.out.println(total);
    }
}

File: sources/kotlin/04_array_sum.kt
// task 04 array_sum — expected output: 499999500000
// build: kotlinc 04_array_sum.kt -include-runtime -d prog.jar    run: java -jar prog.jar    [native build: kotlinc-native -opt -o prog 04_array_sum.kt    native run: ./prog]

fun main() {
    val n = 1000000
    val array = IntArray(n)
    for (i in 0 until n) {
        array[i] = i
    }
    var total = 0L
    for (i in 0 until n) {
        total += array[i]
    }
    println(total)
}
